### ISSUE TITLE
Introduce join provisioning

### ISSUE THREAD

**Issue by johanstokking** [TheThingsNetwork/arduino-device-lib#50]
When provisioning the device during runtime (e.g. through serial, WiFi or Bluetooth), we need to be able to set the `AppEUI` and `AppKey`, without having to join, or let it reset before join.

Proposed API:

```
bool provision(const byte appEui[8], const byte appKey[16]);
```

This method does a `mac set appeui`, `mac set deveui` and `mac save`, see [command reference](http://ww1.microchip.com/downloads/en/DeviceDoc/40001784B.pdf).

Then, you can call just `join()` in `setup()` or `loop()`, that uses the AppEUI and AppKey from memory, and thus only takes retries, as in #39

**Comment by FokkeZB** [comment]
As part of this change `bool join(const byte appEui[8], const byte appKey[16])` should first call `provision(const byte appEui[8], const byte appKey[16])` and then `bool join()`

**Comment by FokkeZB** [comment]
Closed by #59
